task/copy.go

```diff
@@ -213,64 +206,48 @@ func (t *CopyStreamTask) run() {
 	partSize := int64(constants.DefaultPartSize)
 	t.SetPartSize(partSize)
 
-	t.GetScheduler().Sync(initTask)
+	t.GetScheduler().Sync(ctx, initTask)
 	if t.GetFault().HasError() {
 		return
 	}
 	t.SetSegment(initTask.GetSegment())
 
 	offset, part := int64(0), 0
 	for {
+		it := NewInitSegmentStream(t)
+		t.GetScheduler().Sync(ctx, it)
+		if t.GetFault().HasError() {
+			return
+		}
+
 		x := NewCopyPartialStream(t)
+		x.SetSize(it.GetSize())
+		x.SetContent(it.GetContent())
 		x.SetOffset(offset)
 		x.SetIndex(part)
-		t.GetScheduler().Async(x)
+		t.GetScheduler().Async(ctx, x)
 
-		if x.GetDone() {
+		if it.GetDone() {
 			break
 		}
 
-		offset += x.GetSize()
+		offset += it.GetSize()
 		part++
 	}
 
 	t.GetScheduler().Wait()
 	if t.GetFault().HasError() {
 		return
 	}
-	t.GetScheduler().Sync(NewSegmentCompleteTask(initTask))
+	t.GetScheduler().Sync(ctx, NewSegmentCompleteTask(initTask))
 }
 
-func (t *CopyPartialStreamTask) new() {
```

Comment:
Split `new()` method apart into `InitSegmentStreamTask.run()` to implement run with context. Because we cannot handle ctx in `new()` method.